plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConversionContext.kt

```diff
@@ -7,13 +7,21 @@ package org.jetbrains.kotlin.formver.conversion
 
 import org.jetbrains.kotlin.fir.expressions.FirExpression
 import org.jetbrains.kotlin.fir.symbols.impl.FirNamedFunctionSymbol
+import org.jetbrains.kotlin.fir.declarations.FirFunction
+import org.jetbrains.kotlin.fir.symbols.impl.FirFunctionSymbol
+import org.jetbrains.kotlin.fir.symbols.impl.FirRegularClassSymbol
 import org.jetbrains.kotlin.fir.types.ConeKotlinType
+import org.jetbrains.kotlin.formver.embeddings.ClassEmbedding
 import org.jetbrains.kotlin.fir.types.resolvedType
 import org.jetbrains.kotlin.formver.embeddings.MethodSignatureEmbedding
 import org.jetbrains.kotlin.formver.embeddings.TypeEmbedding
 
 interface ProgramConversionContext {
-    fun add(symbol: FirNamedFunctionSymbol): MethodSignatureEmbedding
+
+    fun <D : FirFunction> add(symbol: FirFunctionSymbol<D>): MethodSignatureEmbedding
```

Comment:
I think it's cleaner to just use `FirFunctionSymbol<*>` and remove the type parameter as long as you don't explicitly need it (same in overriden methods).